File: frontend/src/Models/DeliveryPoint.ts
export interface TrustedPoint extends DeliveryPoint {
  id: string;
}

export interface DeliveryPoint {
  id: string | undefined;
  location: google.maps.LatLngLiteral;
  hours: OpeningHours;
  address: string;
}

export interface DisplayPoint extends DeliveryPoint {
  distanceText: string;
  timeText: string;
}

export type Time = {
  hour: number;
  minute: number;
};
type OpeningHour = {
  openingTime: Time;
  closingTime: Time;
};
export type OpeningHours = Record<number, OpeningHour>;

File: api/ApiSource/Points/DeliveryPoint.cs
namespace ApiSource.Points {
    public abstract class DeliveryPoint {
        public Coords Location { get; }
        public string Address { get; }
        public List<TimeSlot> Hours { get; }

        protected DeliveryPoint(Coords location, string address, List<TimeSlot> hours) {
            Location = location;
            Address = address;
            Hours = hours;
        }
    }
}
